crates/common/flockfile/src/unix.rs

```diff
@@ -83,6 +84,15 @@ impl Flockfile {
                     source: err,
                 })?;
 
+        // Convert the PID to a string
+        let pid_string = format!("{}", Pid::this());
+
+        // Write the PID to the lock file
+        write(file.as_raw_fd(), pid_string.as_bytes()).map_err(|err| FlockfileError::FromNix {
+            path: path.clone(),
+            source: err,
+        })?;
```

Comment:
This is done too soon. The pid must be written to the file *after* the lock has been successfully acquired.

Otherwise, a concurrent process erases the pid of the lock owner.

```
$ tedge-mapper c8y >/dev/null 2>&1 &
[1] 188552

# The PID of the mapper is correctly set
$ cat /var/run/lock/tedge-mapper-c8y.lock
188552

# But if the a concurrent mapper is spawned
$ tedge-mapper c8y
2023-03-24T15:24:12.373980146Z ERROR flockfile::unix: Another instance of tedge-mapper-c8y is running.
2023-03-24T15:24:12.374092121Z ERROR flockfile::unix: Lock file path: /run/lock/tedge-mapper-c8y.lock
Error: Couldn't acquire file lock.
Caused by:
    EAGAIN: Try again

# the concurrent process is stopped
# but the PID has been changed - unexpectedly
$ cat /var/run/lock/tedge-mapper-c8y.lock
188628%                                  
```

Reply:
> Fixed. I have got one question regarding this PR, there is also `unistd::ftruncate` function offered by the `nix` crate that allows to truncate the file to zero bytes before writing the PID. I am wondering whether there will be situations where the content of our lockfile may not be empty even though it should?

No this is useless as already done on open with the write option without the append option.